DuckDuckGo.Fluent.Plugin/DuckDuckGoSearchApp.cs

```diff
@@ -0,0 +1,228 @@
+        {
+            return _applicationInfo;
+        }
+
+        public async IAsyncEnumerable<ISearchResult> SearchAsync(SearchRequest searchRequest,
+            [EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            string searchedTag = searchRequest.SearchedTag;
+            string searchedText = searchRequest.SearchedText;
+
+            if (!VerifySearchedTerms(ref searchedText, ref searchedTag)) yield break;
+
+            if (searchedTag.Equals(QrTag))
+            {
+                DuckDuckGoSearchResult duckGoSearchResult = await GetQrImage(searchedText);
+                if (duckGoSearchResult != null)
+                    yield return duckGoSearchResult;
+            }
+            else
+            {
+                string url = GetEndpointUrl(searchedText);
+                var channel = Channel.CreateUnbounded<DuckResult>();
+                using var httpClient = new HttpClient();
+                httpClient.DefaultRequestHeaders.UserAgent.TryParseAdd(UserAgentString);
+
+                _ = httpClient.GetFromJsonAsync<Root>(url, SerializerOptions, cancellationToken).ContinueWith(
+                    async task =>
+                    {
+                        if (task.IsCompletedSuccessfully && task.Result != null)
+                        {
+                            Root root = task.Result;
+
+                            // Get Answers
+                            DuckResult answers = GetAnswers(root, searchedText);
+                            if (answers != null) await channel.Writer.WriteAsync(answers, cancellationToken);
+
+                            // Get Definitions if available
+                            DuckResult dictionary = GetDictionary(root, searchedText);
+                            if (dictionary != null) await channel.Writer.WriteAsync(dictionary, cancellationToken);
+
+                            // Get Abstract in Text form
+                            DuckResult abstractResult = GetAbstract(root, searchedText);
+                            if (abstractResult != null)
+                                await channel.Writer.WriteAsync(abstractResult, cancellationToken);
+
+                            // External Links associated with search like Official Website etc.
+                            if (root?.Results != null)
+                                foreach (DuckResult duckResult in root.Results.Select(variableTopic => new DuckResult
+                                {
+                                    Info = variableTopic.Text, SourceUrl = variableTopic.FirstUrl,
+                                    ResultType = "Links",
+                                    SearchedText = searchedText, SearchResultType = ResultType.SearchResult, Score = 7
+                                }))
+                                    await channel.Writer.WriteAsync(duckResult, CancellationToken.None);
+
+                            // Internal Links associated with Search.
```

Comment:
You should follow the convention above and get result from factory and then write it to channel, instead of passing the channel to external method